### PR TITLE
new mutation: updatePostStatus

### PR DESCRIPTION
**Pull request by Schparky**
Some of the logic and/or tests may not be in their final form. Feel free to comment on that.

### CODE REVIEW

File: application/gqlgen/schema.graphql [hunk 1]
```diff
@@ -230,3 +231,8 @@ input LocationInput {
     latitude: Float
     longitude: Float
 }
+
```
[the remark this answers is not in the document]
Comment (Schparky) [reply application/gqlgen/schema.graphql]: I thought about marking it a deprecated. What do you think, @wcjr?

File: application/gqlgen/post.go [hunk 2]
```diff
@@ -487,3 +487,30 @@ func (r *mutationResolver) UpdatePost(ctx context.Context, input postInput) (*mo
...
+		"user":      cUser.Uuid,
+		"oldStatus": post.Status,
+		"newStatus": input.Status.String(),
+	}
+	if !cUser.CanUpdatePostStatus(post, input.Status.String()) {
+		return nil, reportError(ctx, errors.New("not allowed to change post status"),
+			"UpdatePostStatus.NotAllowed", extras)
+	}
+
+	post.Status = input.Status.String()
```
[the remark this answers is not in the document]
Comment (Schparky) [reply application/gqlgen/post.go]: Yes, good call, I added the next line on a different day and didn't see the redundancy.

File: application/gqlgen/schema.graphql (hunk 1 above)
[the remark this answers is not in the document]
Comment (wcjr) [reply application/gqlgen/schema.graphql]: no need to deprecate, I'll put the change in when it's merged

File: application/models/post.go [hunk 3]
```diff
@@ -551,12 +551,12 @@ func (p *Post) IsEditable(user User) (bool, error) {
...
+	return p.isPostEditable(), nil
 }
 
-// isStatusEditable defines which posts statuses can be edited. It does not use the receiver `p'.
-func (p *Post) isStatusEditable(status string) bool {
-	switch status {
+// isPostEditable defines at which states can posts be edited.
+func (p *Post) isPostEditable() bool {
+	switch p.Status {
 	case PostStatusOpen:
```
[the remark this answers is not in the document]
Comment (wcjr) [reply application/models/post.go]: like that!

**Pull request merged**